examples/sprite_camera_follow/main.rs

```diff
@@ -0,0 +1,160 @@
+extern crate amethyst;
+
+use amethyst::{
+    assets::{AssetStorage, Loader},
+    core::{Parent, Transform, TransformBundle},
+    ecs::{Component, Entity, Join, NullStorage, Read, ReadStorage, System, WriteStorage},
+    input::{InputBundle, InputHandler},
+    prelude::*,
+    renderer::{
+        Camera, ColorMask, DisplayConfig, DrawSprite, MaterialTextureSet, Pipeline, PngFormat,
+        Projection, RenderBundle, SpriteRender, SpriteSheet, SpriteSheetFormat, SpriteSheetHandle,
+        Stage, Texture, TextureMetadata, ALPHA,
+    },
+    utils::application_root_dir,
+};
+
+#[derive(Default)]
+struct Player;
+
+impl Component for Player {
+    type Storage = NullStorage<Self>;
+}
+
+struct MovementSystem;
+
+impl<'s> System<'s> for MovementSystem {
+    type SystemData = (
+        ReadStorage<'s, Player>,
+        WriteStorage<'s, Transform>,
+        Read<'s, InputHandler<String, String>>,
+    );
+
+    fn run(&mut self, (players, mut transforms, input): Self::SystemData) {
+        let x_move = input.axis_value("entity_x").unwrap();
+        let y_move = input.axis_value("entity_y").unwrap();
+
+        for (_, transform) in (&players, &mut transforms).join() {
+            transform.translation.x += x_move as f32 * 5.0;
+            transform.translation.y += y_move as f32 * 5.0;
+        }
+    }
+}
+
+fn load_sprite_sheet(world: &mut World) -> SpriteSheetHandle {
+    let res_path = format!(
+        "{}/examples/sprite_camera_follow/resources",
+        application_root_dir()
+    );
+    let texture_handle = {
+        let loader = world.read_resource::<Loader>();
+        let texture_storage = world.read_resource::<AssetStorage<Texture>>();
+        loader.load(
+            format!("{}/Circle.png", res_path),
+            PngFormat,
+            TextureMetadata::srgb_scale(),
+            (),
+            &texture_storage,
+        )
+    };
+    let texture_id = 0;
+    let mut material_texture_set = world.write_resource::<MaterialTextureSet>();
+    material_texture_set.insert(texture_id, texture_handle);
+
+    let loader = world.read_resource::<Loader>();
+    let sprite_sheet_store = world.read_resource::<AssetStorage<SpriteSheet>>();
+    loader.load(
+        format!("{}/Circle.ron", res_path),
+        SpriteSheetFormat,
+        texture_id,
+        (),
+        &sprite_sheet_store,
+    )
+}
+
+// Initialize a sprite as a reference point at a fixed location
+fn init_reference_sprite(world: &mut World, sprite_sheet: &SpriteSheetHandle) -> Entity {
+    let mut transform = Transform::default();
+    transform.translation.x = 100.0;
+    transform.translation.y = 0.0;
+    let sprite = SpriteRender {
+        sprite_sheet: sprite_sheet.clone(),
+        sprite_number: 1,
+        flip_horizontal: false,
+        flip_vertical: false,
+    };
+    world.create_entity().with(transform).with(sprite).build()
```

Comment:
Little nitpick here: could you add a second reference point so that is clear to the user that the player is moving, and not the reference point? Also maybe a background could help on that, do you think it would be interesting?

Reply:
I think a background could be good, though it might take a bit to create. Is a background just a very large sprite in this case, or is there a better way to do it?

Reply:
I would have went for a very large sprite so that the code does not become too complex.

Reply:
Aight, I'll add it now.

Reply:
I went ahead and added a simple background sprite, and added a little more documentation for some of the "gotchas" I found, like sprite ordering and transparency.